server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -470,6 +470,25 @@ public boolean isRemoteStoreNode() {
         return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
     }
 
+    /**
+     * Returns whether the node is a remote data node.
+     *
+     * @return true if the node contains remote store data attributes, false otherwise
+     */
+    public boolean isRemoteDataNode(){
```

Comment:
```suggestion
    public boolean isDataRemoteBacked(){
```

We can explore other name suggestions too

Reply:
I would prefer `isRemoteDataNode` .